src/lib/utitlities/synchronization.ts

```diff
@@ -0,0 +1,230 @@
+): SnippetStore {
+  return {
+    subscribe: localStore.subscribe,
+    async move(movingSnippet: Snippet, sourceFolderId: string, destinationFolderId: string): Promise<void> {
+      await localStore.move(movingSnippet, sourceFolderId, destinationFolderId)
+      if (await remoteStore.isAvailable()) {
+        await remoteStore.move(movingSnippet, sourceFolderId, destinationFolderId)
+      }
+    },
+    async remove(id: string): Promise<void> {
+      await localStore.remove(id)
+      if (await remoteStore.isAvailable()) {
+        await remoteStore.remove(id)
+      }
+    },
+    async upsert(snippet: Snippet): Promise<void> {
+      await localStore.upsert(snippet)
+      if (await remoteStore.isAvailable()) {
+        await remoteStore.upsert(snippet)
+      }
+    },
+    async clone(snippet: Snippet): Promise<void> {
+      await localStore.clone(snippet)
+      if (await remoteStore.isAvailable()) {
+        await remoteStore.clone(snippet)
+      }
+    },
+    async clearAll(): Promise<void> {
+      await localStore.clearAll()
+      if (await remoteStore.isAvailable()) {
+        await remoteStore.clearAll()
+      }
+    },
+    migrationStateStore: localStore.migrationStateStore
+  }
+}
+
+/**
+ * Create one store that combine the data of local and remote folders. It also
+ * "auto merge" the data following "last write wins" rule.
+ *
+ * In more details:
+ *
+ * - If one folder exists within the local store but not within the remote
+ *   store, then the folder would be created within the remote store, and vice
+ *   versa.
+ * - If the folder within both stores, then the "newer" folder (one that has
+ *   greater `updatedAt` value) would be used and be written to both stores.
+ *
+ * This strategy is much simpler than `createHigherSnippetsStore`'s strategy
+ * basing on an assumption: the snippets' data is important and conflicts should
+ * be resolved instead of letting data be overwritten.
+ * */
+export function createHigherFoldersStore(
+  localStore: LocalFoldersStore,
+  remoteStore: RemoteFoldersStore,
+): LocalFoldersStore {
+  const store = derived(
+    [localStore, remoteStore],
+    ([localFolders, remoteFolders]) => {
+      const allFolders: {[key: string]: DisplayFolder} = {}
+      for (const localFolder of localFolders) {
+        allFolders[localFolder.id] = localFolder
+      }
+      for (const remoteFolder of remoteFolders) {
+        if (remoteFolder.id in allFolders) {
+          const localFolder = allFolders[remoteFolder.id]
+          if (remoteFolder.updatedAt > localFolder.updatedAt) {
+            allFolders[remoteFolder.id] = remoteFolder
+            localStore.upsert(remoteFolder).then()
+          } else {
+            remoteStore.upsert(localFolder).then()
+          }
+        } else {
+          allFolders[remoteFolder.id] = remoteFolder
+          localStore.upsert(remoteFolder).then()
+        }
+      }
+      return Object.values(allFolders)
+    }
+  )
```

Comment:
Only in "real-life" testing that I realize this is a horrible idea: `localStore.upsert` is going to trigger changes again and again, and put the whole tab to frozen.